amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -59,67 +63,106 @@ class SearchPage extends React.Component<SearchPageProps, SearchPageState> {
-        name: "Ash Ketchum",
-        offboarded: false,
-        office: "city",
-        role: "Pokemon Trainer",
-        start_date: "10-10-2014",
-        team_name: "Team Red",
-        title: "Pikachu",
-      },{
-        type: ResourceType.user,
-        active: false,
-        birthday: '10-10-1990',
-        department: "Department",
-        email: "mail@email.com",
-        first_name: "Gary",
-        github_username: "Github User",
-        id: 12345,
-        last_name: "Oak",
-        manager_email: "manager@company.com",
-        name: "Gary Oak",
-        offboarded: false,
-        office: "city",
-        role: "Pokemon Trainer",
-        start_date: "10-10-2014",
-        team_name: "Team Blue",
-        title: "Eevee",
-      }],
-      total_results: 1,
+      results: [],
+      total_results: 0,
     }
   };
 
   constructor(props) {
     super(props);
 
-    this.handlePageChange = this.handlePageChange.bind(this);
-    this.updateQueryString = this.updateQueryString.bind(this);
+    this.state = {
+      selectedTab: ResourceType.table,
+    };
   }
 
   componentDidMount() {
     this.props.getPopularTables();
 
     const params = qs.parse(window.location.search);
-    const searchTerm = params['searchTerm'];
-    const pageIndex = params['pageIndex'];
+    const { searchTerm, pageIndex, selectedTab} = params;
+
+    this.updateSelectedTab(selectedTab);
     if (searchTerm && searchTerm.length > 0) {
       const index = pageIndex || '0';
-      this.props.executeSearch(searchTerm, index);
+      this.props.searchAll(searchTerm, this.getSearchOptions(index));
     }
   }
 
+  updateSelectedTab = (newTab) => {
+    switch(newTab) {
+      case ResourceType.table:
+      case ResourceType.user:
+        this.setState({ selectedTab: newTab });
+        break;
+      case ResourceType.dashboard:
+        break;
+    }
+  };
+
+  getSearchOptions = (pageIndex) => {
+    return {
+      dashboardIndex: (this.state.selectedTab === ResourceType.dashboard) ? pageIndex : 0,
+      userIndex: (this.state.selectedTab === ResourceType.user) ? pageIndex : 0,
+      tableIndex: (this.state.selectedTab === ResourceType.table) ? pageIndex : 0,
+    };
+  };
+
+  getPageIndex = (tab) => {
+    switch(tab) {
+      case ResourceType.table:
+        return this.props.tables.page_index;
+      case ResourceType.user:
+        return this.props.users.page_index;
+      case ResourceType.dashboard:
+        return this.props.dashboards.page_index;
+    }
+    return 0;
+  };
+
+  onSearchBarSubmit = (searchTerm: string) => {
+    this.props.searchAll(searchTerm, this.getSearchOptions(0));
+    this.updatePageUrl(searchTerm, this.state.selectedTab,0);
+  };
+
+  onPaginationChange = (pageNumber) => {
+    // subtract 1 : pagination component indexes from 1, while our api is 0-indexed
+    const index = pageNumber - 1;
+
+    this.props.searchResource(this.state.selectedTab, this.props.searchTerm, index);
```

Comment:
Have we tested this? Will the page re-render on pagination change? If I understand correctly, the component will only re-render when it's state changes. Right now the only piece of state is `selectedTab`, which doesnt get updated in this flow.

Reply:
`searchResource` will update the `searchState` object in the appropriate tab. This will get passed back into the `SearchPage` as props.

Tested an verified.

Reply:
Today I learned the component will re-render when the store updates its props...and that those props don't *need* to be on the component state. (though correct me if you've found anything that says otherwise). 

Some components I wrote will eventually need some cleanup if you want to be on the lookout for those -- we definitely have some components where I passed values to the component state which I now realize don't need to be on the component state because they are only updated from the store -- and not via any internal component interactions. 😓